src/Mod/Spreadsheet/App/Sheet.cpp

```diff
@@ -104,7 +105,15 @@ Sheet::Sheet()
 
 Sheet::~Sheet()
 {
-    clearAll();
+    try {
+        clearAll();
+    }
+    catch (boost::regex_error&) {
+        // Don't let an exception propagate out of a destructor (calls terminate())
+        Base::Console().Error(
+            "clearAll() resulted in an exception when deleting the spreadsheet : %s\n",
+            *pcNameInDocument);
+    }

```

Comment:
Should we have a generic catch as well if the implementation changes and we throw something else in clearAll()?

Reply:
Sure: it might make sense to *only* have a generic exception, it's not like I'm giving any extra information knowing that it's a boost::regex_error, and this *should* be quite rare.

Reply:
Guess a fix for this would look like this then:
```suggestion
    try {
        clearAll();
    }
    catch (...) {
        // Don't let an exception propagate out of a destructor (calls terminate())
        Base::Console().Error(
            "clearAll() resulted in an exception when deleting the spreadsheet : %s\n",
            *pcNameInDocument);
    }
```